northstar/src/runtime/fork/init/mod.rs

```diff
@@ -166,59 +287,20 @@ impl Init {
                     .expect("Failed to spawn child process");
 
                     // close fds
-                    drop(console_stream);
-                    unistd::close(stdout).expect("Failed to close stdout");
-                    unistd::close(stderr).expect("Failed to close stderr");
+                    match out {
+                        Out::Pipe { stdout, stderr } => {
+                            unistd::close(stdout).expect("Failed to close stdout");
+                            unistd::close(stderr).expect("Failed to close stderr");
```

Comment:
Replace the `RawFd` in `Out` with something that closes on drop. This manual fd handling is as dangerous as a `malloc` ;-)